sporkfish/searcher/move_ordering.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+
+import chess
+
+from ..board.board import Board
+
+
+class MoveOrder(ABC):
```

Comment:
I suspect MoveOrder design might be abit wonky but will work in the end. The problem is going to be how we combine these into a production move ordering, see the example from [Rustic](https://rustic-chess.org/search/ordering/killers.html#:~:text=A%20%22killer%20move%22%20is%20a,moves%20the%20opponent%20may%20have.) combining MVV-LVA and killer moves:

```
pub fn score_moves(ml: &mut MoveList, tt_move: ShortMove, refs: &SearchRefs) {
    for i in 0..ml.len() {
        let m = ml.get_mut_move(i);
        let mut value: u32 = 0;

        if m.captured() != Pieces::NONE {
            value = MVV_LVA_OFFSET + MVV_LVA[m.captured()][m.piece()] as u32;
        } else {
            let ply = refs.search_info.ply as usize;
            let mut i = 0;
            while i < MAX_KILLER_MOVES && value == 0 {
                let killer = refs.search_info.killer_moves[i][ply];
                if m.get_move() == killer.get_move() {
                    value = MVV_LVA_OFFSET - ((i as u32 + 1) * KILLER_VALUE);
                }
                i += 1;
            }
        }

        m.set_sort_score(value);
    }
}
```

Aggregation of different move orders is done under specific conditions (i.e. there exists captures in this case for MVV-LVA), so we can't do it "generically". I think we can do is keep the current inheritance design and use it for testing  individual and have a production SporkfishHeuristic - similar to the Rustic one - with our own twists of course. Let me know what you think

Reply:
@yibeili - let us know too, your area coming up :)

Reply:
As discussed offline, current design should be ok - I left it as is.

I need to think more about this.

Reply:
My stupid idea is just for our production move ordering class to inherit from all our individual move ordering classes and conditionally call them as needed